tests/metadata_stores/test_resolve_update.py

```diff
@@ -38,15 +42,339 @@
 from metaxy.metadata_store.clickhouse import ClickHouseMetadataStore
 from metaxy.metadata_store.duckdb import DuckDBMetadataStore
 from metaxy.models.plan import FeaturePlan
-from metaxy.provenance.types import HashAlgorithm
+from metaxy.provenance.types import HashAlgorithm, LazyIncrement
 
 # Type alias for feature plan output
 FeaturePlanOutput = tuple[
     FeatureGraph, Mapping[FeatureKey, type[BaseFeature]], FeaturePlan
 ]
 
 
-# ============= FEATURE GRAPH CONFIGURATIONS =============
+def verify_lazy_sql_frames(store: MetadataStore, lazy_result: LazyIncrement):
```

Comment:
I don't think you need to add this: we already make an assertion with `df.implementation` (either in this test or golden provenance tests); but I will take a closer look

Reply:
removed